File: pe-math/src/main/java/org/saar/maths/noise/SpreadNoise2f.java
package org.saar.maths.noise;

public class SpreadNoise2f implements Noise2f {

    private final int division;
    private final Noise2f noise2f;

    public SpreadNoise2f(int division, Noise2f noise2f) {
        this.division = division;
        this.noise2f = noise2f;
    }

    @Override
    public float noise(float x, float y) {
        return this.noise2f.noise(x / this.division, y / this.division);
    }
}

File: pe-math/src/main/kotlin/org/saar/maths/noise/SpreadNoise2f.kt
package org.saar.maths.noise

class SpreadNoise2f(private val division: Float, private val noise2f: Noise2f) : Noise2f {

    override fun noise(x: Float, y: Float) = this.noise2f.noise(x / this.division, y / this.division)
}
